src/v/datalake/datalake_manager.cc

```diff
@@ -370,4 +371,40 @@ void datalake_manager::stop_translator(const model::ntp& ntp) {
     });
 }
 
+ss::future<uint64_t> datalake_manager::disk_usage() {
+    const auto path = config::node().datalake_staging_path();
+
+    if (!co_await ss::file_exists(path.string())) {
+        co_return 0;
+    }
+
+    chunked_vector<std::filesystem::path> files;
+    co_await directory_walker::walk(
+      path.string(), [&files, path](const ss::directory_entry& de) {
+          if (de.type == ss::directory_entry_type::regular) {
+              files.push_back(path / std::filesystem::path(de.name));
+          }
+          return ss::now();
+      });
+
+    uint64_t total = 0;
+    co_await ss::max_concurrent_for_each(
+      files.begin(),
+      files.end(),
+      config::shard_local_cfg().space_management_max_log_concurrency(),
+      [&total](const std::filesystem::path& path) {
+          return ss::file_size(path.string())
+            .then([&total](uint64_t size) { total += size; })
+            .handle_exception([path](std::exception_ptr eptr) {
+                vlog(
+                  datalake_log.debug,
```

Comment:
nit: would it be too noisy to log this at warn level? Seems like it could be helpful in an incident to see right away if it's causing space management to behave badly

Reply:
we could log errors at warn level. except that it is normal for the occasional file_does_not_exist error to occur (file listing, stating, and removal all race), so we'd probably want to pluck that one out and either log it at INFO or DEBUG. wdyt?

Reply:
Ah yea good call, file_does_not_exist at debug sounds good

Reply:
fixed